department-of-veterans-affairs/caseflow
# HunJerBAH/APPEALS-10393: Add Error Validation to Split Appeal 'Continue' Button

Resolves [APPEAL-10393](https://vajira.max.gov/browse/APPEALS-10393)

### Description
Added a feature that will disable the 'Continue' button on the Create Split Appeal page when all the request issue checkboxes are selected or none of the request issue checkboxes are selected. 

### Acceptance Criteria
- [ ] If the user selects NO appeal issues to split, then the continue button will be disabled.

- [ ] If the user selects ALL appeal issues to split, then the continue button will be disabled.

- [ ] If the user selects appeal issues to split and then deselects all those issues, then the continue button will be disabled.

### Testing Plan
1. Go to Caseflow login and sign in as a Supervisory Senior Council member (SPLT members).
![image](https://user-images.githubusercontent.com/99915461/197872862-f2b6051e-9eb8-459c-813f-d4808fcad3a9.png)
2. Go to **your queue**, **search cases** and docket number **221016-1** to find an appeal. Click on the docket number to go to the case details page
3. On the case details page, click the **correct issues** link to open the split appeal process.
![image](https://user-images.githubusercontent.com/99915461/197873456-f2a38904-8edb-4605-8944-702fe6d9329b.png)
4. If there are no issues on the appeal, click **add issue** and fill out the form. Do this twice so there are two issues on the appeal. Click **save** to save the issues, and then click **correct issues** to come back to the edit issues page. If there are 2 issues on the appeal already, skip this step.
![image](https://user-images.githubusercontent.com/99915461/197873589-5b5d3492-f4b7-4d32-ac7f-74979c6a6edc.png)
![image](https://user-images.githubusercontent.com/99915461/197873693-b3aaa95d-c9f9-450d-a11d-8f65c3e17c65.png)
![image](https://user-images.githubusercontent.com/99915461/197873560-4fb3a42b-1b18-4298-ad29-1ab3a6fb8c15.png)

5. Click **split appeal** to navigate to the create split page.
6. On the create split page, notice that the **continue** button is originally disabled. Click the dropdown menu, and select a reason. The **continue** button should still be disabled.
![image](https://user-images.githubusercontent.com/99915461/197874055-2fcbbce0-7b0f-4dc4-9031-2619dcde114a.png)
![image](https://user-images.githubusercontent.com/99915461/197874168-c93f6bf0-17b6-44cc-8603-09fc2ffdef9f.png)
7. Select one of the issues. The **continue** button should now be active. 
![image](https://user-images.githubusercontent.com/99915461/197874262-27f709f7-f80e-4a5e-afaa-2b8ffee492b0.png)

9. Select both issues on the appeal. The **continue** button should now be disabled.
![image](https://user-images.githubusercontent.com/99915461/197874361-a5749bec-3f98-491b-a90b-34cd2a431fee.png)

10. Deselect both issues. The **continue** button should be active when 1 issues is selected, but disabled when both aren't selected. 
![image](https://user-images.githubusercontent.com/99915461/197874489-4685685f-1003-4989-8f6d-2863fc3b15e6.png)
![image](https://user-images.githubusercontent.com/99915461/197874516-e958609d-d864-4946-8e50-fcac81fa92a8.png)

11. Select one issue and click the **continue** button. On the review page, click the **back** button. Repeat steps 6-10 again to make sure the button works as intended.
